nixos/modules/services/web-apps/vikunja.nix

```diff
@@ -0,0 +1,235 @@
+        };
+        path = mkOption {
+          type = types.str;
+          default = "${cfg.stateDir}/vikunja.db";
+          description = "Path to the sqlite3 database file.";
+        };
+        createDatabase = mkOption {
+          type = types.bool;
+          default = true;
+          description = "Whether to create a local database automatically.";
+        };
+      };
+      corsOrigins = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        description = "A list of origins allowed in the CORS Header to access the API.";
+      };
+      mailer = {
+        enabled = mkOption {
+          type = types.bool;
+          default = false;
+          description = "Whether mailer should be enabled.";
+        };
+        host = mkOption {
+          type = types.str;
+          default = "";
+          description = "SMTP host address.";
+        };
+        port = mkOption {
+          type = types.int;
+          default = 587;
+          description = "SMTP host port.";
+        };
+        username = mkOption {
+          type = types.str;
+          default = "";
+          description = "SMTP username.";
+        };
+        password = mkOption {
+          type = types.str;
+          default = "";
+          description = ''
+            SMTP User password <option>config.database.user</option>.
+            Warning: this is stored in cleartext in the Nix store!
+            Use an environment file with <option>environmentFiles</option> instead.
+          '';
+        };
+        fromemail = mkOption {
+          type = types.str;
+          default = "";
+          description = "The default from address when sending emails.";
+        };
+        forcessl = mkOption {
+          type = types.bool;
+          default = false;
+          description = "By default, vikunja will try to connect with starttls, use this option to force it to use ssl.";
+        };
+      };
+    };
+  };
+  config = lib.mkIf cfg.enable {
+
+    systemd.services.vikunja-api = {
+      description = "vikunja-api";
+      after = [ "network.target" ] ++ lib.optional usePostgresql "postgresql.service" ++ lib.optional useMysql "mysql.service";
+      wantedBy = [ "multi-user.target" ];
+      path = [ cfg.package-api ];
+
+
+      serviceConfig = {
+        Type = "simple";
+        User = cfg.user;
+        Group = "vikunja";
+        WorkingDirectory = cfg.stateDir;
+        ExecStart = "${cfg.package-api}/bin/api";
+        Restart = "always";
+        EnvironmentFile = cfg.environmentFiles;
+      };
+    };
+
+    services.nginx.virtualHosts."${cfg.frontendHostname}" = {
+      locations = {
+        "/" = {
+          root = cfg.package-frontend;
+          tryFiles = "try_files $uri $uri/ /";
+        };
+        "~* ^/(api|dav|\.well-known)/" = {
+          proxyPass = "http://localhost:3456";
+          extraConfig = ''
+            client_max_body_size 20M;
+          '';
+        };
+      };
+    };
```

Comment:
At what point do you want an explanation or an option? For me it would also be ok to serve over SSL by default, but I don't think everyone wants that.

For example, the nextcloud nixos module doesn't have SSL enabled by default either. There you have to activate it yourself, for example with:
```
services.nginx.virtualHosts."${config.services.nextcloud.hostName}" = {
  enableACME = true;
  forceSSL = true;
};
```

Reply:
I think a little documentation would go a long way. The `nextcloud` module has an entire section in the manual. Maybe just a few lines that inform the sysadmin they are responsible for configuring `nginx` and a practical example of what most users should do

Reply:
Do you have a recommendation where I should add the documentation?

Reply:
I think adding it to the `description` of `setupNginx` is probably a good spot. Keep it short and simple and I think it will provide a large amount of value for a small amount of effort :+1:

Reply:
Am I understanding correctly that you want me to configure nginx without SSL in the module and add some lines on how to enable SSL in the `setupNginx` `description`?

Reply:
Sure, that sounds good to me.

Reply:
I added something. You could have a look whether that's good

Reply:
Perfect!